transport/src/main/java/io/netty/channel/ChannelHandlerAdapter.java

```diff
@@ -55,6 +56,10 @@ public boolean isSharable() {
         Boolean sharable = cache.get(clazz);
         if (sharable == null) {
             sharable = clazz.isAnnotationPresent(Sharable.class);
+            if (!sharable) {
```

Comment:
Is this correct? Child class could be nonsharable, while it extends `@Sharable`.

Reply:
@doom369 not sure I understand you correctly.. but `@Sharable` uses `@Inherited` so if the super-class is sharable the child class should be as well.

Reply:
@normanmaurer missed that. Anyway, in that case:

```
 * Indicates that an annotation type is automatically inherited.  If
 * an Inherited meta-annotation is present on an annotation type
 * declaration, and the user queries the annotation type on a class
 * declaration, and the class declaration has no annotation for this type,
 * then the class's superclass will automatically be queried for the
 * annotation type.  This process will be repeated until an annotation for this
 * type is found, or the top of the class hierarchy (Object)
 * is reached.  If no superclass has an annotation for this type, then
 * the query will indicate that the class in question has no such annotation.
```

```
    @ChannelHandler.Sharable
    private static class A {
    }
    private static class B extends A {
    }
    @Test
    public void aaa() {
        System.out.println(B.class.isAnnotationPresent(ChannelHandler.Sharable.class)); //prints true
    }
```

Reply:
@doom369 agree... so you are saying we do not need the extra "query" ? If so I think you are right... @kiril-me did you check if you really need it ?

Reply:
@normanmaurer this check is useful only for the case when the class implements interface marked with `@Sharable`. I don't think this is real use case, however, for at least 1 more additional test case should be added.

Reply:
@kiril-me can you do this ?

Reply:
@normanmaurer There are differences between annotation of the class and inner anonymous class annotation. 

> @Inherited JavaDoc
> Note that this meta-annotation type has no effect if the annotated
> type is used to annotate anything other than a class.  Note also
> that this meta-annotation only causes annotations to be inherited
> from superclasses; annotations on implemented interfaces have no
> effect.

As @doom369 mentioned, isAnnotationPresent(Sharable.class) looking for @Sharable annotation starting from current class till Object.

But this scenario doesn't work for isAnnotationPresent, because @Inherit does not apply directly to the class:
```
ChannelInboundHandler handler = new @Sharable ChannelInboundHandlerAdapter() {
      @Override
      public void channelRead(ChannelHandlerContext context, Object message) {
           context.write(message);
      }
};
```
Tests checking this. If you comment out following code, test will fails:
```
if (!sharable) {       
                AnnotatedType annotatedType = clazz.getAnnotatedSuperclass();
                sharable = annotatedType.isAnnotationPresent(Sharable.class);
 }
```

Reply:
Thanks for the explanation.